kitsune/middleware.py

```diff
@@ -0,0 +1,199 @@
+            return response
+
+        urlparts = request.path_info.split('/')
+        try:
+            if not urlparts[1].startswith('t5'):
+                return response
+        except KeyError:
+            return response
+
+
+        # Oh boy, Lithium URL. Let's see if we can turn things around.
+
+        # Cache first
+        cache_key = hashlib.sha1(request.path_info.encode('utf-8')).hexdigest()
+        cached_redirect = cache.get(cache_key)
+        if cached_redirect:
+            return HttpResponseRedirect(cached_redirect)
+
+        if request.path_info in self.extra_pages:
+            return HttpResponseRedirect(self.extra_pages[request.path_info])
+
+        if urlparts[2] in self.product_pages:
+            return HttpResponseRedirect(self.product_pages[urlparts[2]])
+
+        path = request.path_info
+        url = 'https://secure02.lithium.com.edgekey.net' + path
+        try:
+            response = requests.get(
+                url,
+                headers={'Host': 'support.mozilla.org'},
+                timeout=3,
+            )
+        except requests.exceptions.Timeout:
+            sentry_client.captureException()
+            return handle404(request)
+
+        try:
+            response.raise_for_status()
+        except requests.exceptions.RequestException:
+            sentry_client.captureException()
+            return handle404(request)
+        else:
+            for content, prop in self.meta_regex.findall(response.content):
+                try:
+                    content = content.strip().decode('utf-8')
+                    prop = prop.strip().decode('utf-8')
+                except UnicodeError:
+                    break
+
+                if prop == 'og:title':
+                    # order of lookup is important
+                    try:
+                        bar = Document.objects.get(title=content)
+                    except (Document.DoesNotExist, Document.MultipleObjectsReturned):
+                        pass
+                    except:
+                        sentry_client.captureException()
+                    else:
+                        url = bar.get_absolute_url() + '?cache=no'
+                        cache.set(cache_key, url, None)
+                        return HttpResponseRedirect(url)
+
+                    if '-' in content:
+                        product, title = content.split('-', 1)
+                        product = product.strip()
+                        title = title.strip()
+
+                        title = TOPICS.get(title, title)
+                        product = PRODUCTS.get(product, product)
+
+                        try:
+                            bar = Topic.objects.get(title=title, product__title=product)
+                        except (Topic.DoesNotExist, Topic.MultipleObjectsReturned):
+                            pass
+                        except:
+                            sentry_client.captureException()
```

Comment:
This is an interesting pattern. The goal is to record the error in sentry, without interrupting the view, right? That's pretty clever.